### PR TITLE
Unify pre-commit config across repos

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Addresses WordPress/openverse#316 by @dhruvkb

## Description

This PR uses the new pre-commit config introduced in WordPress/openverse#330 and fixes the code so that the lint checks pass.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/catalog/api/controllers/search_controller.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def _unmasked_query_end(page_size, page):
 
 def _paginate_with_dead_link_mask(
     s: Search, page_size: int, page: int
-) -> Tuple[int, int]:
+) -> tuple[int, int]:
```
Comment (obulat) [inline api/catalog/api/controllers/search_controller.py:55]: Interesting that it replaces the types this way!

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -19,6 +21,7 @@ repos:
         args:
           - --remove
       - id: pretty-format-json
+        exclude: package(-lock)?\.json
```
Comment (obulat) [inline .pre-commit-config.yaml:24]: Would this affect `tsconfig.json` in any way?

**Review by obulat — APPROVED** [review]
Cleaner look is always welcome :)

File: .pre-commit-config.yaml (hunk 2 above)
Comment (dhruvkb) [reply .pre-commit-config.yaml:24]: This isn't being synced with the frontend yet. There are some repo-specific hooks in the frontend so syncing them centrally is not feasible (yet).

File: api/catalog/api/utils/help_text.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from typing import Iterable
+from collections.abc import Iterable
```
Comment (krysal) [inline api/catalog/api/utils/help_text.py:1]: Interesting change 😮

File: .git-blame-ignore-revs [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# Add pyupgrade
+5158c579d0c8eeb6e9372b30a03ce48f53459958
```
[the remark this answers is not in the document]
Comment (dhruvkb) [reply .git-blame-ignore-revs:2]: You're completely right, that'll not work!

**Pull request merged**